src/main/java/com/mparticle/kits/SingularKit.java

```diff
@@ -0,0 +1,235 @@
+            } catch (Exception unableToGetDDLTimeout) {
+            }
+        }
+        config = new SingularConfig(singularKey, singularSecret);
+        config.withDDLTimeoutInSec(DDL_HANDLER_TIMEOUT_SEC);
+        config.withDDLHandler(this);
+        if (MParticle.getInstance().getEnvironment() == MParticle.Environment.Development) {
+            config.withLoggingEnabled();
+            config.withLogLevel(Log.DEBUG);
+        }
+        Singular.init(context, config);
+        List<ReportingMessage> messages = new ArrayList<ReportingMessage>();
+        messages.add(new ReportingMessage(this, ReportingMessage.MessageType.APP_STATE_TRANSITION, System.currentTimeMillis(), null));
+        return messages;
+    }
+
+    @Override
+    public List<ReportingMessage> setOptOut(boolean b) {
+        return null;
+    }
+
+    @Override
+    public String getName() {
+        return "Singular";
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityCreated(Activity activity, Bundle bundle) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityStarted(Activity activity) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityResumed(Activity activity) {
+        Singular.onActivityResumed();
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityPaused(Activity activity) {
+        Singular.onActivityPaused();
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityStopped(Activity activity) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivitySaveInstanceState(Activity activity, Bundle bundle) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityDestroyed(Activity activity) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> leaveBreadcrumb(String s) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logError(String s, Map<String, String> map) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logException(Exception e, Map<String, String> map, String s) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logEvent(MPEvent mpEvent) {
+        List<ReportingMessage> messages = new LinkedList<ReportingMessage>();
+        String eventName = mpEvent.getEventName();
+        Map eventInfo = mpEvent.getInfo();
+        if (eventInfo != null) {
+            JSONObject params = new JSONObject(eventInfo);
+            eventStatus = Singular.event(eventName, params.toString());
+        } else {
+            eventStatus = Singular.event(eventName);
+        }
+        if (eventStatus) {
+            messages.add(ReportingMessage.fromEvent(this, mpEvent));
+        } else {
+            messages.add(new ReportingMessage(this, ReportingMessage.MessageType.ERROR, System.currentTimeMillis(), null));
```

Comment:
This will end up showing in the mParticle UI that we forwarded an error message TO Singular, not that we receive an error FROM singular. Please remove this line - in the case of `eventStatus` being `false`, just do not add the ReportingMessage to the `messages` list.